### PR TITLE
Fix message split in #split_message

### PR DESCRIPTION
**Pull request by captainSV**
# Summary

---

## Changed
#split_message now automatically splits messages on the last ` ` character, instead of just automatically filling out all 2000 characters, because that was causing issues with breaking links or making words commit sepuku.

### CODE REVIEW

File: lib/discordrb/data/channel.rb [hunk 1]
```diff
@@ -380,6 +380,7 @@ def send_multiple(content)
     # Useful for sending long messages, but be wary of rate limits!
     def split_send(content)
       send_multiple(Discordrb.split_message(content))
+      nil
```
Comment (Daniel-Worrall) [inline lib/discordrb/data/channel.rb]: Why is this here?
This is breaking behaviour as it currently returns an Array of the split content.
If you're using this in a CommandBot, remember to end your block with a nil instead of expecting this message to return nil

Comment (z64) [reply lib/discordrb/data/channel.rb]: It's not breaking because this method has no documented return value.

File: lib/discordrb.rb [hunk 2]
```diff
@@ -52,6 +52,13 @@ def self.split_message(msg)
     # being the character limit, otherwise just return an array with one element
     ideal_ary = ideal.length > CHARACTER_LIMIT ? ideal.chars.each_slice(CHARACTER_LIMIT).map(&:join) : [ideal]
 
+    # If there is no space to logically split the message on in the message (No spaces at all), it stays the same, otherwise
+    # cut string on index of the last space, and put everything after that space in the next index
+    # Note: please find a way to be more inclusive of 'where to logical split message', /\b/ captures the end of string
+    0.upto(ideal_ary.length-2) { |x| ideal_ary[x].rindex(/ /).nil? ? (ideal_ary[x] = ideal_ary[x])
+    : (ideal_ary[x], ideal_ary[x+1] = ideal_ary[x][0..ideal_ary[x].rindex(/ /)],
+    (ideal_ary[x][ideal_ary[x].rindex(/ /)..-1] + ideal_ary[x+1])) }
```
Comment (z64) [inline lib/discordrb.rb]: Sorry, this isn't acceptable. Please write this code out in longer form - this is unnecessarily difficult to read and maintain :(

Comment (captainSV) [reply lib/discordrb.rb]: Right, so i expanded the ternary and commented about what each line does.

File: lib/discordrb.rb [hunk 3]
```diff
@@ -55,9 +55,15 @@ def self.split_message(msg)
     # If there is no space to logically split the message on in the message (No spaces at all), it stays the same, otherwise
     # cut string on index of the last space, and put everything after that space in the next index
     # Note: please find a way to be more inclusive of 'where to logical split message', /\b/ captures the end of string
-    0.upto(ideal_ary.length-2) { |x| ideal_ary[x].rindex(/ /).nil? ? (ideal_ary[x] = ideal_ary[x])
-    : (ideal_ary[x], ideal_ary[x+1] = ideal_ary[x][0..ideal_ary[x].rindex(/ /)],
-    (ideal_ary[x][ideal_ary[x].rindex(/ /)..-1] + ideal_ary[x+1])) }
+
+    0.upto(ideal_ary.length-2) do |x|
+      if ideal_ary[x].rindex(/ /).nil? #if there is no space to break on, just leave it be
```
Comment (swarley) [inline lib/discordrb.rb]: Can we use `' '`, `/\s/` or `/\b/` here?

Comment (captainSV) [reply lib/discordrb.rb]: /\s/ is probably best in my opinion
also change it to `unless` instead of `if/else`, I was being dumb when I wrote that

File: lib/discordrb/data/channel.rb (hunk 1 above)
Comment (omnilord) [reply lib/discordrb/data/channel.rb]: It may not be documented but it most certainly would be a breaking change for anyone already using this function and using the split message array for logging purposes.

Comment (captainSV) [reply lib/discordrb/data/channel.rb]: I don't know why this is showing up as a change, considering it's already in the repo as of #629 when it was accepted

Comment (Daniel-Worrall) [reply lib/discordrb/data/channel.rb]: Seems you're behind with commits is why.

File: lib/discordrb.rb [hunk 4]
```diff
@@ -48,9 +48,9 @@ def self.split_message(msg)
     # Find the largest element that is still below the character limit, or if none such element exists return the first
     ideal = joined.max_by { |e| e.length > CHARACTER_LIMIT ? -1 : e.length }
 
-    # If it's still larger than the character limit (none was smaller than it) split it into slices with the length
-    # being the character limit, otherwise just return an array with one element
-    ideal_ary = ideal.length > CHARACTER_LIMIT ? ideal.chars.each_slice(CHARACTER_LIMIT).map(&:join) : [ideal]
+    # If it's still larger than the character limit (none was smaller than it) split it into the largest chunk without
+    # cutting words apart, breaking on the nearest space within character limit, otherwise just return an array with one element
+    ideal_ary = ideal.length > 2000 ? ideal.split(/(.{,2000}\b+?)/).reject(&:empty?) : [ideal]
```
Comment (greenbigfrog) [inline lib/discordrb.rb]: Why is `CHARACTER_LIMIT` hardcoded to 2000 here?!
```suggestion
    ideal_ary = ideal.length > CHARACTER_LIMIT ? ideal.split(/(.{,CHARACTER_LIMIT}\b+?)/).reject(&:empty?) : [ideal]
```

Comment (captainSV) [reply lib/discordrb.rb]: Honestly? The way i was doing my algorithmic testing i didn't feel like also having a while line for declaring character limit for, the literal one line of code in the test, so i changed it to 2000.
and, evidently, forgot to change it back before pushing to my repo. That's my bad.
